### ISSUE TITLE
Autocomplete for in-text citation entries 

### ISSUE THREAD

**Issue by acebulsk** [quarto-dev/quarto-nvim#42]
Maybe this is already available but I could not find out how to enable it. Is it possible to have auto-complete work after typing the `@` symbol to search entries listed in the bibliography entered within the yaml header.

i.e.

```
I am searching for the citation @cit....
```

would autocomplete `@cit` from the bibliography below:

```
---
title: "Notes on Reading List"
format: 
  pdf:
    toc: true 
bibliography: '../papers/bibtex/library.bib'
---
```

Also it would be nice to disable spellcheck after the `@` symbol.

**Comment by jmbuhr** [comment]
I explain treesitter queries for syntax highlighting (which includes spell checking since a couple of month ago) here: https://youtu.be/hSWjh1waiso?t=358

**Comment by acebulsk** [comment]
Ah I found my problem the auto-complete function only works if the bibliography path is **not** surrounded by quotes. So, this did not work:

```
bibliography: '../papers/bibtex/library.bib'
```

but this does:

```
bibliography: ../papers/bibtex/library.bib
```

Looking forward to the modifications to the spellcheck to exclude `@cite`. Thanks for your work on this!
Cheers.

**Comment by jmbuhr** [comment]
That's a bug indeed then. Would you mind opening an issue in https://github.com/jmbuhr/cmp-pandoc-references so that I can keep track of it?

**Comment by jmbuhr** [comment]
ok, I have experimented a bit with this (https://github.com/jmbuhr/quarto-nvim-kickstarter/commit/55b62ae3a7cff76b30cb1b3243c69aa023a6cb8d), the results are in the latest version of the kickstarter config.

However, there are two caveats:
I was only able to disable spellchecking for urls in links and `[...]` references, but not inline references `@...`. The reason is that `@someIdentifier`  doesn't have it's own node type in the markdown treesitter grammar. So while I am able to match an `@`, adding a highlighting or nospell to the query would match apply to the whole node, which is a whole paragraph in this case. This is obviously not what we want.
Secondly, in playing around with this I noticed that the additional legacy syntax highlighting also messes with spelling regions and changes highlighting in weird ways when spellcheck is enabled, which makes me re-evaluate weather it should be enabled by default in the config (this line: https://github.com/jmbuhr/quarto-nvim-kickstarter/blob/55b62ae3a7cff76b30cb1b3243c69aa023a6cb8d/lua/plugins/treesitter.lua#LL10C16-L10C16).

Both of these issues should be addressed when we have our own quarto grammar because it would allow us to define proper captures for citations and make the legacy syntax highlighting completely obsolete.
